pkg/appmgmt/general/general_test.go

```diff
@@ -526,3 +527,269 @@ func TestGetOwnerReferences(t *testing.T) {
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+		},
+	})
+
+	// app02 pods, running in queue02 and namespace02
+	// 2 pods are having applicationID and queue name specified
+	// both 2 pods are pending
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app02pod0001",
+			Namespace: ns02,
+			Labels: map[string]string{
+				constants.LabelApplicationID: app02,
+				constants.LabelQueueName:     queue02,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+		},
+	})
+
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app02pod0002",
+			Namespace: ns02,
+			Labels: map[string]string{
+				constants.LabelApplicationID: app02,
+				constants.LabelQueueName:     queue02,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+		},
+	})
+
+	// app03 pods, running in queue02 and namespace02
+	// 2 pods do not have label applicationID specified, but have spark-app-selector
+	// both 2 pods are allocated
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app03pod0001",
+			Namespace: ns01,
+			Labels: map[string]string{
+				constants.SparkLabelAppID: app03,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+			NodeName:      "some-node",
+		},
+	})
+
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app03pod0002",
+			Namespace: ns02,
+			Labels: map[string]string{
+				constants.SparkLabelAppID: app03,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+			NodeName:      "some-node",
+		},
+	})
+
+	// app04 pods, running in queue01 and namespace01
+	// app04 has 2 pods which only has annotation yunikorn.apache.org/app-id specified
+	// both 2 pods are allocated
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app04pod0001",
+			Namespace: ns01,
+			Annotations: map[string]string{
+				constants.AnnotationApplicationID: app04,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+			NodeName:      "some-node",
+		},
+	})
+
+	mockedPodLister.AddPod(&v1.Pod{
+		TypeMeta: apis.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: apis.ObjectMeta{
+			Name:      "app04pod0002",
+			Namespace: ns01,
+			Annotations: map[string]string{
+				constants.AnnotationApplicationID: app04,
+			},
+		},
+		Spec: v1.PodSpec{
+			SchedulerName: constants.SchedulerName,
+			NodeName:      "some-node",
+		},
+	})
```

Comment:
This is a duplicate from app04pod0001 and does not test anything new, should be removed.